clause/where.go

```diff
@@ -165,8 +170,8 @@ func (not NotConditions) Build(builder Builder) {
 			builder.WriteString("NOT ")
 			e, wrapInParentheses := c.(Expr)
 			if wrapInParentheses {
-				sql := strings.ToLower(e.SQL)
-				if wrapInParentheses = strings.Contains(sql, "and") || strings.Contains(sql, "or"); wrapInParentheses {
```

Comment:
There is not need to change check with space.

Just use `strings.Contains(sql, "AND") || strings.Contains(sql, "OR")` is better

Reply:
Sorry, @huacnlee, but it seems that it is necessary, as GORM has it coded in this manner in multiple places. So I'm just wanting to make it consistent with the rest of the functional code.

References for:
" OR ":
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L38
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L134
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L137

" AND ":
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L29
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L113
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L116
https://github.com/go-gorm/gorm/blob/19ac396a22668e2cdbd77a262de84478787989d0/clause/where.go#L159

The reason that adding the spaces on both ends of the `and` and `or` as requirement is because, if one is scanning a string that contains "score" as a field, as in the test case, this logic will enclose the field in Parentheses. Which is unnecessary, and only occurs when the type is `clause.Expr`.

e.g.,
```
clause.And(clause.Expr{SQL: "`score` <= ?", Vars: []interface{}{100}, WithoutParentheses: false})
```

For background, my team had found this bug when we created our sqlMocks and had to write exception mocks when testing with a db field with the name of `category`. And yes, this is for selfish reasons, if you apply my fix to master, I can simplify my mocks and not have an odd `if field name contains 'or' or 'and' enclose it in Parentheses`.

Thanks in Advance for your understanding.

Reply:
I see.

Reply:
@huacnlee @daheige can we proceed and have this merged into master for a future release?